File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    n = len(sequence)
    left = 0
    cursum = 0
    best = [0, len(sequence)-1]
    
    for right in range(n):
        cursum += sequence[right]
        
        while cursum > k and left < right:
            cursum -= sequence[left]
            left += 1
        if cursum == k:
            # print("left right: " + str(left) + " " + str(right))
            if best[1]-best[0] > right - left or (best[1]-best[0] == right - left and best[0] > left):
                best = [left, right]
    return best

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        
        int[] left_right =  {0, 0};
        int sums = sequence[0];
        int[] best = {0, 1000001};
        
        
        while (left_right[0] <= sequence.length-1 && left_right[1] <= sequence.length-1) {

            // System.out.println(sums);
            // System.out.println(left_right[0] + " " + left_right[1]);
            if (sums == k) {
                if (best[1]-best[0] > left_right[1]-left_right[0]) {
                    best = left_right.clone();
                }
                if (left_right[1] == sequence.length - 1) break;
                left_right[1] += 1;
                sums += sequence[left_right[1]];
            } else if (sums > k) {
                sums -= sequence[left_right[0]];
                
                if (left_right[0] == left_right[1]) {
                    // System.out.println("--끝--");
                    break;
                }
                left_right[0] += 1;
                // System.out.println("----");
            } else if (sums < k) {
                if (left_right[1] == sequence.length-1) break;
                left_right[1] += 1;
                sums += sequence[left_right[1]];
            }
        }
        return best;
    }
}